[Wong Pei Xian] iP

trxe opened the pull request:
# DukeAgain


```
Hello from
 ____        _        
|  _ \ _   _| | _____ 
| | | | | | | |/ / _ \
| |_| | |_| |   <  __/
|____/ \__,_|_|\_\___|
```


If you are cool with a ~buggy~ TUI project with weird input formatting, use DukeAgain as a:

1. terminal-based task manager
2. easy to use
3. simple task management options

Download **Duke 0.1** from _**Releases**_ and run `java -jar duke.jar` on your terminal in the same directory as the download. Make sure you've installed Java Runtime Environment on your device (available [here](https://www.oracle.com/java/technologies/javase-jre8-downloads.html)).

-----

## Quick start documentation
The following are the recognized formats for 3 different types of tasks (items in round brackets are optional:

- `todo [description]`
- `deadline [description] /by [(yyyy-MM-dd) (HH:mm)] (/ details)`
- `event [description] /by [(yyyy-MM-dd) (HH:mm)] ~ [(yyyy-MM-dd) (HH:mm)] (/ details)`
- `find [keyword(s)]`
- ℹ️ `help` to learn other commands! 

-----

## Features to be added

- [ ] Sort by time
- [ ] More flexible formatting
- [ ] Reminders

> _"yea boi"_
> ~ internet guy

Commit:
Add Gradle support

trxe committed:
Level-1

trxe committed:
Level-2

trxe committed:
Level-3

trxe committed:
Level-4

trxe committed:
Level-4; methods corresponding to user commands are now individual private methods and not lambdas

trxe committed:
Level-4; more flexibility in Deadline.java and Event.java inputs

trxe committed:
A-TestUiTesting

trxe committed:
Level-5; Duke.java now catches errors

trxe committed:
Level-5; tested

trxe committed:
minor bug fixes and formatting

trxe committed:
Add enum in comments

trxe committed:
Remove enum

trxe committed:
Add basic file-reading functionality

trxe committed:
basic filewriting functionality

trxe committed:
Remove duke.txt from .gitignore

trxe committed:
Clean

trxe committed:
Level-7; Save

trxe committed:
Level-8; Dates and Times

trxe committed:
Merge branch 'add-gradle-support' of https://github.com/trxe/ip.git into branch-Level-7

trxe committed:
Prepare for merge

trxe committed:
Merge branch 'master' into branch-Level-8

trxe committed:
Fix merge bugs

trxe committed:
A-MoreOOP; Create classes Ui, Parser, Database, Storage; DukeDateTime and help documentation in progress

trxe committed:
DukeDateTime handles time

trxe committed:
Create packages duke, duke.utils, duke.tasks

trxe committed:
Add 3 new unit tests under JUnit

trxe committed:
Generate JAR

trxe committed:
Add JavaDoc documentation

trxe committed:
A-CodingStandard; Adhere to CS2103 Coding Standard

trxe committed:
Find setup

trxe committed:
Add find unit test

trxe committed:
Small styling adjustment

trxe committed:
Merge branch 'branch-A-CodingStandard'

trxe committed:
Merge branch 'master' into branch-Level-9

trxe committed:
Added remaining JavaDoc

trxe committed:
Add full Gradle support as needed

trxe committed:
Add GUI; problem with displaying long text (truncation)

trxe committed:
Commit master

trxe committed:
CheckStyle

trxe committed:
Merge branch 'master' into branch-A-CheckStyle

trxe committed:
Merge branch 'master' into branch-A-Gradle

trxe committed:
Merge branch 'master' into branch-Level-10

wonyeji commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    private Record todo(String args) throws DukeException {
```
Perhaps you could use a verb like `addTodo` instead of `todo` for the method name? Same for the other addTask methods below!

trxe replied:
Personally I think its clearer in my code setup that the command the user issues matches the name of private function in the `Parser` class that corresponds to it, so I don't thnk I will be changing these names

wonyeji commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+        }
+        return new Record("Here are the matching tasks in your list:\n " + filtered);
+    }
+}
```
There seems to be a missing newline at the end of the file! I noticed it for a few other files as well.

trxe replied:
Newlines are added by the static method `Ui.reply`, thanks though!

wonyeji commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @return String representing this deadline in database format.
+     */
+    public String toDB() {
```
```suggestion
    public String toDatabase() {
```

trxe replied:
Changed to `toDatabaseFormat`, thanks!

wonyeji commented on src/main/java/duke/utils/Record.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * Creates a new record.
+     *
+     * @param message
```
I think these may be missing their descriptions?

wonyeji commented on src/main/java/duke/utils/Record.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * Getter for terminate.
+     *
+     * @return
```
This is missing the description as well!

wonyeji commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+     * Takes in a command, and if command is found, passes to it the remaining arguments.
+     *
+     * @param input
+     * @return
```
Missing return description!

wonyeji commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    private String sizeMsg() {
```
Perhaps you could use a verb for this method name? Something like `getTaskSize()`

wonyeji commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,192 @@
+
+public class Parser {
+
+    private final TaskList db;
```
Perhaps a more meaningful name like `dataBase` could be used? 👍

wonyeji commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,113 @@
+            }
+            boolean done = Integer.parseInt(args[1]) == 1;
+            switch (args[0]) {
+            case "T":
```
Rather than just the char here, perhaps you could add a char constant to the different Task classes to enhance readability? Also to make it easier to edit in the future when task types with overlapping char representations are created! :~)

trxe replied:
Thank you for the suggestion, will implement

wonyeji reviewed (commented):
Overall, brilliant job! Just a few nits to fix, otherwise your code is amazing! 💯

trxe committed:
Minor updates from comments

trxe committed:
Help command: help command prints out full manual.

Users are likely to only be looking for help of a certain command.
Additionally the help command doesn't format properly in GUI when
it's too long.

"help" prints full help message and "help (cmd)" prints syntax and
usage of that command.

Let's
* improve the help function by giving 2 variants of help
* "help": prints whole dictionary manual
* "help (cmd)": prints manual entry of (cmd).

trxe committed:
A-Assertions: Assert files present

trxe committed:
Filled in missing assertion

trxe committed:
A-CodeQuality: Minor improvements to code quality and implementation of 'bye' command

trxe committed:
Fix error in saving and loading data

trxe committed:
Use stream to generate the database text for 'list' command

trxe committed:
Merge pull request #1 from trxe/branch-A-Assertions

A-Assertions

trxe committed:
Merge pull request #2 from trxe/branch-A-CodeQuality

A-CodeQuality: Minor improvements to code quality and implementation …

trxe committed:
Merge branch 'master' into branch-A-Streams

trxe committed:
Merge pull request #3 from trxe/branch-A-Streams

A-Streams

trxe committed:
Update TaskList

trxe committed:
Merge branch 'branch-A-Streams'

trxe committed:
C-Help: Add help manual and help command

trxe committed:
Remove redundant imports

trxe committed:
Edit minor formatting issue

trxe committed:
Fix tagging for BCD-Extension

trxe committed:
Fix message truncation problem, change GUI text to monospace

trxe committed:
Clean up text formatting to fit screen

trxe committed:
Add command aliases for todo, deadline and event

trxe committed:
Merge branch 'master' into branch-UI

trxe committed:
Prepare for release, user guide in progress

trxe committed:
Progressing user guide

trxe committed:
Fix bugs with filepaths

trxe committed:
Fix help

trxe committed:
Fix help filepath and event printing bug

trxe committed:
Progress user guide

trxe committed:
Progress user guide

trxe committed:
Progress user guide

trxe committed:
Upload Ui.png

trxe committed:
Upload Ui.png

trxe committed:
Release

trxe committed:
Release

trxe committed:
Fix visual bug on MacOS

trxe committed:
Release

trxe committed:
Release

trxe committed:
Improve code quality

trxe committed:
Update README.md

trxe committed:
Finalise user guide:

trxe committed:
Add help to resources

trxe committed:
Finish first fully working product

trxe committed:
Add Javadoc header comments

trxe committed:
Change Task done attribute to isDone
